app/content/profile/index.js

```diff
@@ -125,7 +138,68 @@ function Index() {
   }
 
   const handleImageClick = () => {
-    console.log("TODO: implement and open an image upload modal!");
+    setIsImageUploadModalVisible(true);
+  };
+
+  // Function to handle closing the image upload modal
+  const hideImageUploadModal = () => {
+    setIsImageUploadModalVisible(false);
+  };
+
+  const firebaseProfileImageUpload = async (uri) => {
+    try {
+      setImageUploading(true);
+      // Fetch the image data from the URI
+      const response = await fetch(uri);
+      const blob = await response.blob();
+
+      // Get a reference to the Firebase Storage instance
+      const storage = getStorage();
+
+      // Create a reference to the storage location where the image will be stored
+      const storageRef = ref(storage, userId);
+
+      // Upload the image to Firebase Storage
+      const snapshot = await uploadBytes(storageRef, blob);
+
+      console.log("Uploaded the image blob to the Firebase storage:", snapshot);
+
+      // Get the download URL for the uploaded image
+      const downloadURL = await getDownloadURL(snapshot.ref);
+      await updateDoc(doc(db, "teams", currentTeamId, "users", userId), {
+        pfp: downloadURL,
+      });
+
+      queryClient.invalidateQueries({ queryKey: ["user", { userId }] });
+      setSnackbarMessage("Successfully uploaded the profile picture!");
+      setImageUploading(false);
+
+      return downloadURL;
+    } catch (error) {
+      console.error("Error uploading image to Firebase:", error);
+      setSnackbarMessage("Error uploading profile picture. Please try again.");
+      setImageUploading(false);
+
+      throw error; // Rethrow the error to handle it at the caller's level if needed
+    }
+  };
+
+  // Function to handle uploading image locally
+  const handleImageUpload = async () => {
+    let imageResult = await launchImageLibraryAsync({
+      mediaTypes: MediaTypeOptions.Images,
```

Comment:
Is it possible to compress the image before uploading it?

Reply:
I don't think so? Is there a need?